### PR TITLE
fix: belongs_to link inside tabs

### PR DESCRIPTION
**Pull request by Paul-Bob**
# Description

Fixes https://github.com/avo-hq/avo/issues/1194

belongs to link fix

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/docs)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by adrianthedev** [comment]
I pushed some changes for the comment resource. It doesn't work for me there. Does it work for you?

Also, can we add tests for that please?

**Review by adrianthedev — CHANGES_REQUESTED** [review]
It doesn't work for me on the comment resource. Does it work for you?

Can we also add tests please?

**Comment by Paul-Bob** [comment]
> It doesn't work for me on the comment resource. Does it work for you?
> 
> Can we also add tests please?

@adrianthedev panel items fixed. I moved the comments fields back because some specs are dependent on them. We can do a system test for this, do you agree?

**Review by adrianthedev — APPROVED** [review]
I'm wondering if we still need a piece of code. Otherwise it's pretty much ready to merge

File: lib/avo/fields/belongs_to_field.rb [hunk 1]
```diff
@@ -58,6 +58,8 @@ module Fields
     # - is_disabled?
 
     class BelongsToField < BaseField
+      attr_accessor :inside_tab
```
[the remark this answers is not in the document]
Comment (Paul-Bob) [reply lib/avo/fields/belongs_to_field.rb:61]: You mean using `target` and give it `_top` when inside a panel or a tab?

[the remark this answers is not in the document]
Comment (adrianthedev) [reply lib/avo/fields/belongs_to_field.rb:61]: Yes! And `nil` by default.

[the remark this answers is not in the document]
Comment (Paul-Bob) [reply lib/avo/fields/belongs_to_field.rb:61]: Sure, that's actually a great suggestion to simplify our problem, when i choose `inside_tab` name I was thinking about the possibility of reusing that boolean for some other contexts but for now we don't need that so I agree on keep it simple ;)

File: lib/avo/concerns/has_fields.rb [hunk 2]
```diff
@@ -277,6 +277,17 @@ def get_items
           end
         end
 
+        # Make sure all tabs panelfull_items are setted as inside tabs
+        panelfull_items.grep(Avo::TabGroup).each do |tab_group|
```
[the remark this answers is not in the document]
Comment (Paul-Bob) [reply lib/avo/concerns/has_fields.rb:281]: Always learning! =))

File: lib/avo/tab_group_builder.rb [hunk 3]
```diff
@@ -15,7 +15,7 @@ def initialize(style: nil)
   end
 
   def field(field_name, **args, &block)
-    parsed = Avo::Dsl::FieldParser.new(id: field_name, order_index: @items_index, **args, &block).parse
+    parsed = Avo::Dsl::FieldParser.new(id: field_name, order_index: @items_index, **args.merge(inside_tab: true), &block).parse
```
[the remark this answers is not in the document]
Comment (Paul-Bob) [reply lib/avo/tab_group_builder.rb:18]: I think we don't, great observation

**Pull request merged**